tests/sparseml/transformers/test_transformers.py

```diff
@@ -0,0 +1,164 @@
+        ]
+    )
+
+
+def _run_inference_onnx(path_onnx, input_data):
+    ort_sess = ort.InferenceSession(path_onnx)
+    model = onnx.load(path_onnx)
+    model_input = OrderedDict([(inp.name, None) for inp in model.graph.input])
+
+    with np.load(input_data) as data:
+        for idx, input_name in enumerate(model_input.keys()):
+            model_input[input_name] = data[data.files[idx]].reshape(1, -1)
+
+    output = ort_sess.run(
+        None,
+        model_input,
+    )
+    return output
+
+
+def _compare_onnx_models(model_1, model_2):
+    major_nodes = [
+        "QLinearMatMul",
+        "Gemm",
+        "MatMul",
+        "MatMulInteger",
+        "Conv",
+        "QLinearConv",
+        "ConvInteger",
+        "QuantizeLinear",
+        "DeQuantizeLinear",
+    ]
+
+    nodes1 = model_1.graph.node
+    nodes1_names = [node.name for node in nodes1]
+    nodes1_count = Counter([node_name.split("_")[0] for node_name in nodes1_names])
+
+    nodes2 = model_2.graph.node
+    nodes2_names = [node.name for node in nodes2]
+    nodes2_count = Counter([node_name.split("_")[0] for node_name in nodes2_names])
+
+    for node in major_nodes:
+        assert nodes1_count[node] == nodes2_count[node]
+
+
+@pytest.mark.parametrize(
+    "model_stub, recipe_present, task",
+    [
+        (
+            "zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned-conservative",  # noqa: E501
+            False,
+            "question-answering",
+        )
+    ],
+    scope="function",
+)
+class TestModelFromZoo:
+    @pytest.fixture()
+    def setup(self, model_stub, recipe_present, task):
+        # setup
+        model = Zoo.load_model_from_stub(model_stub)
+        model.download()
+
+        path_onnx = model.onnx_file.downloaded_path()
+        model_path = model.framework_files[0].dir_path
+
+        yield path_onnx, model_path, recipe_present, task
+
+        # teardown
+        shutil.rmtree(os.path.dirname(model_path))
+
+    def test_load_weights_apply_recipe(self, setup):
+        path_onnx, model_path, recipe_present, task = setup
+        config = AutoConfig.from_pretrained(model_path)
+        model = load_task_model(task, model_path, config)
+
+        assert model
+        assert recipe_present == _is_yaml_recipe_present(model_path)
+        if recipe_present:
+
+            trainer = Trainer(
+                model=model,
+                model_state_path=model_path,
+                recipe=None,
+                recipe_args=None,
+                teacher=None,
+            )
+            applied = trainer.apply_manager(epoch=math.inf, checkpoint=None)
+
+            assert applied
+
+    def test_outputs(self, setup):
+        path_onnx, model_path, recipe_present, task = setup
+        path_retrieved_onnx = export_transformer_to_onnx(
+            task=task,
+            model_path=model_path,
+            onnx_file_name="retrieved_model.onnx",
+        )
+
+        inputs_tar_path = os.path.join(
```

Comment:
should be able to get this from the sample_inputs attribute of sparsezoo Model class